### ISSUE TITLE
[BUG] test case test_setting_priority_class failed in master and v1.5.x

### ISSUE THREAD

**Issue by chriscchien** [longhorn/longhorn#6319]
## Describe the bug (🐛 if you encounter this issue)

In Longhorn master and v1.5.x, setting an non-existent priority class to setting `priority` will get 404 error make test script fail. And  the error message with same action is not the same with v1.4.x.

Daily regression test [result](https://ci.longhorn.io/job/public/job/master/job/sles/job/amd64/job/longhorn-tests-sles-amd64/551/testReport/junit/tests/test_settings/test_setting_priority_class/)

## To Reproduce

1. Perform test case `test_setting_priority_class`
2. Or setting a non-existent priority class into setting `priority-class `

## Expected behavior

Have the same error message make test case have consistent result

## Log or Support bundle

master and v1.5.x error message when set a non-exist priority class
```
(ApiError(...), "404 : \n{'message': '', 'code': 404}")
```
v1.4.x error message when set a non-exist priority class
```
(ApiError(...), '500 : failed to set the setting priority-class with invalid value priority-class-piqd54: failed to get priority class priority-class-piqd54 before modifying priority class setting: priorityclass.scheduling.k8s.io "priority-class-piqd54" not found\n{\'code\': 500, \'detail\': \'\', \'message\': \'failed to set the setting priority-class with invalid value priority-class-piqd54: failed to get priority class priority-class-piqd54 before modifying priority class setting: priorityclass.scheduling.k8s.io "priority-class-piqd54" not found\', \'status\': 500}')
```
longhorn-manager log in master
```
time="2023-07-14T01:54:20Z" level=warning msg="HTTP handling error" error="failed to set the setting priority-class with invalid value notexistpriorityclass1: failed to get priority class notexistpriorityclass1 before modifying priority class setting: priorityclass.scheduling.k8s.io \"notexistpriorityclass1\" not found"
10.42.0.1 - - [14/Jul/2023:01:54:20 +0000] "PUT /v1/settings/priority-class HTTP/1.1" 404 0 "https://54.81.227.171/" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
```

## Environment

 - Longhorn version: master, v1.5.x
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): kubectl
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: v1.27.3+k3s1

## Additional context

N/A

**Comment by yangchiu** [comment]
This can be easily reproduced by setting `Priority Class` from Longhorn UI, set any value to `Priority Class` would cause `404` error.

Not sure but probably related to https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2

**Comment by derekbit** [comment]
> This can be easily reproduced by setting `Priority Class` from Longhorn UI, set any value to `Priority Class` would cause `404` error.
> 
> Not sure but probably related to [longhorn/longhorn-manager@a523288](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2)

Yeah, the side effect of the [PR](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2).
cc @ChanYiLin

**Comment by derekbit** [comment]
IMO, returning 404 rather than 500 in the fix https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2 is correct.
The problem is the error details aren't there. Suggest return more error message in the response as I mentioned in https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2#r121573092.

BTW, we can improve the UI a little bit. In addition to the status code, UI can show the error message in the reply as well.
![image](https://github.com/longhorn/longhorn/assets/12527233/581529ba-4ec9-4a59-a417-881bd08a119d)

**Comment by innobead** [comment]
> Yeah, the side effect of the [PR](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2). cc @ChanYiLin
> 
> IMO, returning 404 rather than 500 in the fix [longhorn/longhorn-manager@a523288](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2) is correct. The problem is the error details aren't there. Suggest return more error message in the response as I mentioned in [longhorn/longhorn-manager@a523288#r121573092](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2#r121573092).
> 
> BTW, we can improve the UI a little bit. In addition to the status code, UI can show the error message in the reply as well. ![image](https://user-images.githubusercontent.com/12527233/253462663-581529ba-4ec9-4a59-a417-881bd08a119d.png)

@derekbit do we still need UI to improve that? I feel we don't need based on your PR which has the same response structure as the current response payload. right?

**Comment by innobead** [comment]
@chriscchien good raise, @yangchiu good root cause, @derekbit good improvement :+1: 

cc @ChanYiLin

**Comment by derekbit** [comment]
> > Yeah, the side effect of the [PR](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2). cc @ChanYiLin
> > IMO, returning 404 rather than 500 in the fix [longhorn/longhorn-manager@a523288](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2) is correct. The problem is the error details aren't there. Suggest return more error message in the response as I mentioned in [longhorn/longhorn-manager@a523288#r121573092](https://github.com/longhorn/longhorn-manager/commit/a5232884268215d2669cfd9bd0e51bfaa4bf69c2#r121573092).
> > BTW, we can improve the UI a little bit. In addition to the status code, UI can show the error message in the reply as well. ![image](https://user-images.githubusercontent.com/12527233/253462663-581529ba-4ec9-4a59-a417-881bd08a119d.png)
> 
> @derekbit do we still need UI to improve that? I feel we don't need based on your PR which has the same response structure as the current response payload. right?

 Nope. I've introduced a new writeErr function. The UI improvement is not necessary.

**Comment by chriscchien** [comment]
Verified **pass** in longhorn master (longhorn-manager `8f1205`)

- Set a non-exist priority class from UI will get below response instead of get 404 error code directly.
![image](https://github.com/longhorn/longhorn/assets/73337386/2b540dad-aa68-46fa-95b3-90dfc0d766d5)
- Test case `test_setting_priority_class` passed on [AMD64](https://ci.longhorn.io/job/public/job/master/job/sles/job/amd64/job/longhorn-tests-sles-amd64/555/testReport/tests/test_settings/) and [ARM64](https://ci.longhorn.io/job/public/job/master/job/sles/job/arm64/job/longhorn-tests-sles-arm64/542/testReport/tests/test_settings/) pipeline
